addons/website/tests/test_base_url.py

```diff
@@ -1,22 +1,26 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import odoo
-import odoo.tests
+from lxml.html import document_fromstring
 
+from odoo.tests import HttpCase, tagged
 
-@odoo.tests.tagged('-at_install', 'post_install')
-class TestBaseUrl(odoo.tests.HttpCase):
-    def test_base_url(self):
-        ICP = self.env['ir.config_parameter']
-        Website = self.env['website']
 
+@tagged('-at_install', 'post_install')
+class TestBaseUrl(HttpCase):
+
+    def setUp(self):
+        super(TestBaseUrl, self).setUp()
+        self.domain = 'https://www.domain.jke'
+        self.website = self.env['website'].browse(1)
```

Comment:
keep  Website.create({'name': 'test base url', 'domain': domain})

Borwse Id instead of search([], limit=1) is a bad idea
And avoid to depend of data inyour test

Reply:
You might also use:
```suggestion
        self.webstie = self.env.ref("website.default_website")
```

Reply:
I'll keep the code to create a new website but set the domain to `'http://' + odoo.tests.HOST` so it is actually selected as the current website when doing `url_open`, which is what I wanted by using `browse(1)`.